File: src/sketchscene.cpp
#include "sketchscene.h"
#include <QGraphicsSceneMouseEvent>
#include <QCursor>
#include <QGraphicsPixmapItem>
#include <QPainterPath>

SketchScene::SketchScene(QObject *parent)
        : QGraphicsScene(parent), bgPixmapItem(nullptr) {

}

SketchScene::SketchScene(const QRectF &sceneRect, QObject *parent)
        : QGraphicsScene(sceneRect, parent), bgPixmapItem(nullptr) {
}

SketchScene::SketchScene(qreal x, qreal y, qreal width, qreal height, QObject *parent)
        : QGraphicsScene(x, y, width, height, parent), bgPixmapItem(nullptr) {
}

SketchScene::~SketchScene() {
}

void SketchScene::mousePressEvent(QGraphicsSceneMouseEvent *event) {
    QGraphicsScene::mousePressEvent(event);
//    qDebug()<<"press..."<<event->scenePos();
}

void SketchScene::mouseMoveEvent(QGraphicsSceneMouseEvent *event) {
    QGraphicsScene::mouseMoveEvent(event);
//    qDebug()<<"move..."<<event->scenePos();
}

void SketchScene::mouseReleaseEvent(QGraphicsSceneMouseEvent *event) {
    QGraphicsScene::mouseReleaseEvent(event);
//    qDebug()<<"release...";
}

QGraphicsPathItem *SketchScene::commitPath() {
    return addPath(currentPath);
}

void SketchScene::addPathMoveTo(const QPointF &p) {
    currentPath.moveTo(p);
}

void SketchScene::addPathLineTo(const QPointF &p) {
    currentPath.lineTo(p);
}

void SketchScene::clearPath() {
//    currentPath.clear();
    currentPath = QPainterPath();
}

void SketchScene::addBackGround(const QSize &size) {
    if (bgPixmapItem != nullptr) {
        if (bgPixmapItem->pixmap().size() == size) {
            return;
        }
        removeItem(reinterpret_cast<QGraphicsItem *>(bgPixmapItem));
    }
    QPixmap tmp(size);
    tmp.fill(Qt::white);
    bgPixmapItem = addPixmap(tmp);
    bgPixmapItem->setZValue(-1);
    bgPixmapItem->setActive(false);
}

void SketchScene::clearAll() {
    clear();
    bgPixmapItem = nullptr;
}

File: python/sketchscene.py
from PySide2.QtCore import Qt
from PySide2.QtGui import QPainterPath, QPixmap
from PySide2.QtWidgets import QGraphicsScene


class SketchScene(QGraphicsScene):
    def __init__(self):
        super(SketchScene, self).__init__()
        self.current_path = QPainterPath()
        self.bg_pixmap_item = None

    def mousePressEvent(self, event):
        super(SketchScene, self).mousePressEvent(event)

    def mouseMoveEvent(self, event):
        super(SketchScene, self).mouseMoveEvent(event)

    def mouseReleaseEvent(self, event):
        super(SketchScene, self).mouseReleaseEvent(event)

    def commitPath(self):
        return self.addPath(self.current_path)

    def addPathMoveTo(self, p):
        self.current_path.moveTo(p)

    def addPathLineTo(self, p):
        self.current_path.lineTo(p)

    def clearPath(self):
        self.current_path = QPainterPath()

    def addBackGround(self, size):
        if self.bg_pixmap_item is not None:
            if self.bg_pixmap_item.pixmap().size() == size:
                return
            self.removeItem(self.bg_pixmap_item)
        tmp = QPixmap(size)
        tmp.fill(Qt.white)
        self.bg_pixmap_item = self.addPixmap(tmp)
        self.bg_pixmap_item.setZValue(-1)
        self.bg_pixmap_item.setActive(False)

    def clearAll(self):
        self.clear()
        self.bg_pixmap_item = None
